Develop

antonio-marasescu opened the pull request:
-Just one last _important_ question: "Do we have free pizza today at the Summer Night: Minute to win it?" =))

antonio-marasescu committed:
Angular Intro

Created the Project

antonio-marasescu committed:
2. Components and Bootstrap

- Added Product-List and Product-Detail component,
-Added Basic Event and Data Binding.
-Added Bootstrap Styling

antonio-marasescu committed:
3. Routing

-Added routing for detail page
-Added routing for products page
-Added default routing to products page
-Added wildcard routing (when no route exists) towards a Page Not Found Component
-Modified Product Service

antonio-marasescu committed:
4.0 Services using HttpClient

-Added getting the products list and product detail from the backend

antonio-marasescu committed:
4.1 Services using HttpClient

-Redesigned using Smart&Dumb Components
-Added Shopping Cart and Basic Functionality + Shopping Cart Service
-Added delete on the Detail Page

antonio-marasescu committed:
5. Forms and Validation

-Added create product component + logic
-Added edit product component + logic
-Used general product form with Validators
-Finished cart logic (increment and decrement)
-Finished http logic (without auth handling)

antonio-marasescu committed:
7. Authorization and Guards

-Added guard and some more UI logic

antonio-marasescu committed:
Pull Request Review Changes

-Solved the Pull Request Issues

antonio-marasescu committed:
Added Animations

-Added background loading animation

antonio-marasescu committed:
Opt 1.0 Material Design

-Started adding material design to the project

TimeKnow committed:
Opt 1.1 Material Design

-Finished adding material design to all the components

antonio-marasescu committed:
OPT-2. Internationalization

-Translated the user interface for Romanian
-Added configurations to start the web server for Romanian Configurations

antonio-marasescu committed:
Opt 3.0 Simple NGRX

- Store Implementation for Users

antonio-marasescu committed:
Opt 3.1

antonio-marasescu committed:
Opt 3.3x Redux

antonio-marasescu committed:
Opt 3.4x Product Store

serban-petrescu commented on online-shop/src/app/background-loading/background-loading.component.ts:
```diff
@@ -0,0 +1,36 @@
+@Component({
+  selector: 'app-background-loading',
+  animations: [
+    trigger('loadingAnimation', [
```
Nice!

serban-petrescu commented on online-shop/src/app/page-product-details/page-product-details.component.ts:
```diff
@@ -0,0 +1,57 @@
+
+  delete(): void {
+    const dialogRef = this.dialog.open(ConfirmationModalComponent, {
+      width: '500px',
```
Small hint: in general it is good to avoid hardcoding sizes. Percentage based sizes or responsive sizes are generally better... In this case though, it is fine like this.

serban-petrescu commented on online-shop/src/app/page-product-edit/page-product-edit.component.ts:
```diff
@@ -0,0 +1,34 @@
+  }
+
+  getProduct(): void {
+    this.productId = +this.route.snapshot.paramMap.get('id');
```
Always use parseInt instead of casting.

serban-petrescu commented on online-shop/src/app/store/actions/product.actions.ts:
```diff
@@ -0,0 +1,115 @@
+import {Action} from '@ngrx/store';
+import {Product} from '../../models/product';
+
+export enum EProductActions {
```
Not sure what the E comes from :) Also, it might be an idea to split up the actions into several files (e.g. one for the detail page and one for the list page).

antonio-marasescu replied:
From an example that I seen I noticed that it follows a coding convention similar to the C# standard('I' at the front for interface, 'E' for enum..)

antonio-marasescu replied:
I'm not sure if it's okay and I was meaning to ask you

serban-petrescu reviewed (approved):
Nice job. The NGRX stuff also looks good (but not finished for products yet).

antonio-marasescu committed:
Opt 3.5x Products

antonio-marasescu committed:
Opt 3.6x Products

antonio-marasescu committed:
Opt 3.7x Products Redux

-Finalized Products Redux

antonio-marasescu committed:
Opt 3.7x Products Redux Done

antonio-marasescu committed:
Opt -6. E2E Testing

-Wrote the test for the user to go to a product detail page (from login)

! Note issue regarding VPN and the E2E tests
